server/consumer.go

```diff
@@ -1332,7 +1369,8 @@ func (o *consumer) setLeader(isLeader bool) {
 		}
 		// Stop any inactivity timers. Should only be running on leaders.
 		stopAndClearTimer(&o.dtmr)
-
+		// Stop any unpause timers. Should only be running on leaders.
+		stopAndClearTimer(&o.uptmr)
```

Comment:
The one on 1085 is to ensure that every call to `updatePausedState` is doing the right thing, on config update, API request etc. If we aren't a leader anymore or we aren't paused anymore, we leave the timer reset.

This one is because we know we're definitely not the leader anymore and therefore definitely don't need the timer.